lumicks/pylake/kymo.py

```diff
@@ -47,17 +38,17 @@ def _default_line_time_factory(self: "Kymo"):
 
             # Special case for no dead time. Is it really no deadtime, or a single line?
             if dead_time == 0 and not np.any(beyond_first_line):
-                raise RuntimeError(single_line_error)
+                return scan_time * infowave._src.dt * ns_to_sec
 
         except ValueError:  # ValueError occurs when beyond_first_line is empty
-            raise RuntimeError(single_line_error)
+            return scan_time * infowave._src.dt * ns_to_sec
```

Comment:
I can't suggest on the whole block, but it seems that the `try ... except` was there mostly to re-throw a more descriptive error. Since we don't need to do that now, we could move the early out up and save some indentation

Personally, I find it cleaner, for lines 35-44:
```
beyond_first_line = infowave.data[start + scan_time :] != InfowaveCode.discard
if not np.any(beyond_first_line):
    return scan_time * infowave._src.dt * ns_to_sec

# Special case for no dead time. Is it really no deadtime, or a single line?
if (dead_time := np.argmax(beyond_first_line)) == 0:
    return scan_time * infowave._src.dt * ns_to_sec
```

I tried it and all of the tests pass

Reply:
Yeah, that's fair. I'd rather check for `len(beyond_first_line)` than using `np.any`, since that's the only case we need to be covering (to prevent `argmax` from getting an empty array).